### PR TITLE
[REVIEW] Update orc reader and writer fuzz tests

### PR DESCRIPTION
**Pull request by galipremsagar**
This PR introduces:

- [x] Fixes to some of the breakages introduced by the latest `pyorc` in using `pyorc.Struct`.
- [x] Adapt to `list` dtype parameter changes introduced previously.
- [x] Misc fixes required for proper fuzz test runs.

### CODE REVIEW

File: python/cudf/cudf/_fuzz_testing/orc.py [hunk 1]
```diff
@@ -67,6 +67,19 @@ def generate_input(self):
             dtypes_meta, num_rows, num_cols = _generate_rand_meta(
                 self, dtypes_list
             )
+            if num_cols == 0:
+                """
+                If a dataframe has no columns, then pyorc writer will throw
```
Comment (vuule) [inline python/cudf/cudf/_fuzz_testing/orc.py:72]: I wondered what the desired behavior is here, i.e. whether ORC as a format supports having no columns.
I think we also have some issues writing empty dataframes to ORC.

**Review by vuule — APPROVED** [review]
LGTM, but it's kind of hard to understand why some changes were made.

File: python/cudf/cudf/_fuzz_testing/orc.py (hunk 1 above)
Comment (galipremsagar) [reply python/cudf/cudf/_fuzz_testing/orc.py:72]: I guess we need to write an `empty struct`. With pyorc we can do it this way:

```python
>>> import pyorc
>>> import pandas as pd
>>> output = open("sample.orc", "wb")
>>> writer = pyorc.Writer(output, pyorc.Struct())
>>> writer.close()
>>> pd.read_orc('sample.orc')
Empty DataFrame
Columns: []
Index: []
```

Looks like I also need to make some code-changes in this PR. I'll update this PR.

**Comment by galipremsagar** [comment]
> LGTM, but it's kind of hard to understand why some changes were made (without the detailed description).

Updated the code-changes with comments, this is ready for a re-review.

**Review by rgsl888prabhu — APPROVED** [review]

File: python/cudf/cudf/_fuzz_testing/utils.py [hunk 2]
```diff
@@ -73,7 +85,8 @@ def _generate_rand_meta(obj, dtypes_list, null_frequency_override=None):
             if null_frequency_override is None
             else null_frequency_override
         )
-        cardinality = obj._rand(obj._max_rows)
+        # `cardinality` has to be atleast 1.
```
Comment (vuule) [inline python/cudf/cudf/_fuzz_testing/utils.py:88]:
```suggestion
        # `cardinality` has to be at least 1.
```

**Review by vuule — APPROVED** [review]
found a typo, LGTM otherwise

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**